File: DSA/AnshulJain_590014934/Day_41_Question_1.py
def remove_adjacent_duplicates(string):
    stack=[]
    for char in string:
        if stack and stack[-1]==char:
            stack.pop()
        else:
            stack.append(char)
    return ''.join(stack)

input_str = input("Enter a string: ")#Taking input from the user 
print(remove_adjacent_duplicates(input_str))
#for example:
# Input: "abbaca"
# Output: "ca"

File: DSA/VaniJain_590012020/Day41Question1.java
import java.util.*;
public class RemoveAdjacentDuplicates 
{
    public static String removeOnce(String s) 
  {
        Stack<Character> stack = new Stack<>();
        for (char ch : s.toCharArray()) 
        {
            if (!stack.isEmpty() && stack.peek() == ch) 
            {
                stack.pop();
            } else 
            {
                stack.push(ch);
            }
        }
        StringBuilder sb = new StringBuilder();
        for (char ch : stack) 
        {
            sb.append(ch);
        }
        return sb.toString();
    }
    public static String removeAllAdjacentDuplicates(String s) 
  {
        String prev;
        String curr = s;
        do 
        {
            prev = curr;
            curr = removeOnce(prev);
       } 
          while (!prev.equals(curr));
        return curr;
    }
    public static void main(String[] args) 
  {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a string: ");
        String input = scanner.nextLine();
        String result = removeAllAdjacentDuplicates(input);
        System.out.println("Result after removing all adjacent duplicates: " + result);
    }
}
